pkgs/test_core/lib/src/runner/vm/platform.dart

```diff
@@ -48,23 +48,53 @@ class VMPlatform extends PlatformPlugin {
 
     _setupPauseAfterTests();
 
-    var receivePort = ReceivePort();
+    MultiChannel outerChannel;
+    var cleanupFns = <void Function()>[];
     Isolate? isolate;
-    try {
-      isolate = await _spawnIsolate(
-          path, receivePort.sendPort, suiteConfig.metadata, platform.compiler);
-      if (isolate == null) return null;
-    } catch (error) {
-      receivePort.close();
-      rethrow;
+    if (platform.compiler == Compiler.exe) {
+      var serverSocket = await ServerSocket.bind('localhost', 0);
+      Process process;
+      try {
+        process =
+            await _spawnExecutable(path, suiteConfig.metadata, serverSocket);
+      } catch (error) {
+        serverSocket.close();
+        rethrow;
+      }
+      process.stdout.map(stdout.add);
```

Comment:
This roughly mirrors the typical isolate behavior, but lmk if you think we should do something different.

Reply:
By default we capture `print` calls in the exe right? So these would only be the actual `stdout.write` calls from the code under test?

I think this would probably be OK, we could also consider treating these more like they were calls to `print` since we _can_, but I don't know how if the inconsistency outweighs the value of making the JSON reporter always work for exe compiled tests.

Reply:
Yes prints still are caught and handled as special messages.

It actually looks like the stdout/stderr messages aren't making it through right now though interestingly, ill look into that.

Reply:
Oh, its because I was using `map` instead of `listen` 🤦 . That is fixed now.

I think the consistency is the most important part here, even though it breaks the json reporter, it is helpful for debugging etc.